src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/PrivateWorkerCraftingRequestResolver.java

```diff
@@ -0,0 +1,163 @@
+import com.minecolonies.api.util.constant.TypeConstants;
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.buildings.AbstractBuilding;
+import com.minecolonies.coremod.colony.buildings.AbstractBuildingWorker;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.TextComponentTranslation;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.minecolonies.api.util.RSConstants.CONST_CRAFTING_RESOLVER_PRIORITY;
+
+/**
+ * A crafting resolver which takes care of 2x2 crafts which are crafted by the requesting worker.
+ */
+public class PrivateWorkerCraftingRequestResolver extends AbstractRequestResolver<IDeliverable>
+{
+    public PrivateWorkerCraftingRequestResolver(@NotNull final ILocation location, @NotNull final IToken<?> token)
+    {
+        super(location, token);
+    }
+
+    @Override
+    public TypeToken<? extends IDeliverable> getRequestType()
+    {
+        return TypeConstants.DELIVERY;
+    }
+
+    @Override
+    public boolean canResolve(@NotNull final IRequestManager manager, final IRequest<? extends IDeliverable> requestToCheck)
+    {
+        if (!manager.getColony().getWorld().isRemote)
+        {
+            final Colony colony = (Colony) manager.getColony();
+            final ILocation requesterLocation = requestToCheck.getRequester().getRequesterLocation();
+            final AbstractBuilding building = colony.getBuilding(requesterLocation.getInDimensionLocation());
+            if(building instanceof AbstractBuildingWorker)
+            {
+                final ItemStack stack = requestToCheck.getRequest().getResult();
+
+                if (ItemStackUtils.isEmpty(stack))
+                {
+                    return false;
+                }
+
+                return ((AbstractBuildingWorker) building).canCraft(stack);
+            }
+        }
+
+        return false;
+    }
+
+    @Nullable
+    @Override
+    public List<IToken<?>> attemptResolve(@NotNull final IRequestManager manager, @NotNull final IRequest<? extends IDeliverable> request)
+    {
+        if (manager.getColony().getWorld().isRemote)
+        {
+            return null;
+        }
+
+        final Colony colony = (Colony) manager.getColony();
+        final ILocation requesterLocation = request.getRequester().getRequesterLocation();
+        final AbstractBuilding building = colony.getBuilding(requesterLocation.getInDimensionLocation());
+        if(canResolve(manager, request) && building instanceof AbstractBuildingWorker)
+        {
+            final ItemStack stack = request.getRequest().getResult();
+            final IRecipeStorage storage = ((AbstractBuildingWorker) building).getFirstFullFillableRecipe(stack);
+            if(storage == null)
+            {
+                return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1168)